restapi/api_test.go

```diff
@@ -1 +1,429 @@
+		checkURL(t, request, "/api/v1/client/trigger")
+		// send response to be tested later
+		writeBody(responseWriter, `this is not proper JSON`)
+	})
+	// close the server when test finishes
+	defer server.Close()
+
+	api := restapi.NewRestAPI(server.URL)
+
+	// perform REST API call against mocked HTTP server
+	_, err := api.ReadListOfTriggers()
+	if err == nil {
+		t.Fatal("Error is expected to be returned")
+	}
+}
+
+// TestReadListOfTriggersResponseError check the method ReadListOfTriggers
+func TestReadListOfTriggersResponseError(t *testing.T) {
+	// start a local HTTP server
+	server := httptest.NewServer(http.NotFoundHandler())
+	// close the server when test finishes
+	defer server.Close()
+
+	api := restapi.NewRestAPI(server.URL)
+
+	// perform REST API call against mocked HTTP server
+	_, err := api.ReadListOfTriggers()
+	if err == nil {
+		t.Fatal("Error is expected to be returned")
+	}
+}
+
+// TestReadListOfConfigurationProfilesEmptyList check the method ReadListOfConfigurationProfiles
+func TestReadListOfConfigurationProfilesEmptyList(t *testing.T) {
+	// start a local HTTP server
+	server := mockedHttpServer(func(responseWriter http.ResponseWriter, request *http.Request) {
+		checkURL(t, request, "/api/v1/client/profile")
+		// send response to be tested later
+		writeBody(responseWriter, `{"status":"ok"}`)
+	})
+	// close the server when test finishes
+	defer server.Close()
+
+	api := restapi.NewRestAPI(server.URL)
+
+	// perform REST API call against mocked HTTP server
+	triggers, err := api.ReadListOfConfigurationProfiles()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(triggers) != 0 {
+		t.Fatal("Expected empty list of triggers")
+	}
+}
+
+// TestReadListOfConfigurationProfilesOneProfile check the method ReadListOfConfigurationProfiles
+func TestReadListOfConfigurationProfilesOneProfile(t *testing.T) {
+	// start a local HTTP server
+	server := mockedHttpServer(func(responseWriter http.ResponseWriter, request *http.Request) {
+		checkURL(t, request, "/api/v1/client/profile")
+		responseAsString := `
+		{
+			"profiles": [{}],
+			"status":"ok"
+		}`
+		// send response to be tested later
+		writeBody(responseWriter, responseAsString)
+	})
+	// close the server when test finishes
+	defer server.Close()
+
+	api := restapi.NewRestAPI(server.URL)
+
+	// perform REST API call against mocked HTTP server
+	triggers, err := api.ReadListOfConfigurationProfiles()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(triggers) != 1 {
+		t.Fatal("Expected list with one trigger only")
+	}
+}
+
+// TestReadListOfConfigurationProfilesErrorStatus check the method ReadListOfConfigurationProfiles
+func TestReadListOfConfigurationProfilesErrorStatus(t *testing.T) {
+	// start a local HTTP server
+	server := mockedHttpServer(func(responseWriter http.ResponseWriter, request *http.Request) {
+		checkURL(t, request, "/api/v1/client/profile")
+		// send response to be tested later
+		writeBody(responseWriter, `{"status":"error"}`)
```

Comment:
As I see, same sequence of `checkURL` and `writeBody` is repeated in most of the cases, so maybe we could do one function with needed parameters?

Reply:
why so?

Reply:
I mean we can define new function with higher complexity and with quite unrelated actions

Reply:
implemented in another commit, thanks a lot @bond95